presto-main/src/main/java/io/prestosql/event/QueryMonitor.java

```diff
@@ -339,6 +374,71 @@ private static QueryIOMetadata getQueryIOMetadata(QueryInfo queryInfo)
         return new QueryIOMetadata(inputs.build(), output);
     }
 
+    private static Multimap<FragmentNode, OperatorStats> extractStatsForTableScan(QueryInfo queryInfo)
```

Comment:
The logic here seems to be kind of non-trivial to me, I was wondering if it'd be worth adding a unit test (for this or  the caller method). Not sure if there's a straightforward way currently.

Reply:
I changed it to the simple form you suggested.